### ISSUE TITLE
Question: is it possible to use selected field's arguments as cypher query constraints?

### ISSUE THREAD

**Issue by kipz** [neo4j-graphql/neo4j-graphql-js#51]
Consider the simple query:

```graphql
query FooBar {
  Foo (name: "alpha") {
    bar (name: "beta"){
      name
    }
  }
}
```
In this scenario, I mostly want to only return the `Foo ` with `name="alpha"` if the `name` field of the `bar: Bar` field of `Foo` equals `beta`, but I actually get all `Foo's` with name `alpha`, and a bunch of nulls where the `name` of `Bar` is not `beta`.

Does anybody know a way to achieve this? 9/10 times this is what I find I need. Is this possible with the `@cypher` directive somehow?

**Comment by johnymontana** [comment]
Hey @kipz we will be sure to support this use case when we add `filter` arguments. In the mean time, you could accomplish this with a `@cypher` directive query that looks something like this:

```
type Query {
  fooBarWithNames(fooName: String, barName: String): [FooBar] @cypher(statement:"""
    MATCH (f:Foo)-[:BAR]->(b:Bar) WHERE f.name = $fooName AND b.name = $barName
    RETURN f
  """
}
```

**Comment by jameslouiz** [comment]
@kipz You could use the APOC [conditionals](https://neo4j-contrib.github.io/neo4j-apoc-procedures/#conditionals)

**Comment by johnymontana** [comment]
Hey @kipz we've released a `filter` argument in v2.5.0 just now that should address your use case. See the docs here: https://grandstack.io/docs/neo4j-graphql-js.html#filtering


Your example would look something like this:
```
query FooBar {
  Foo (filter: {name: "alpha", bar_some: {name: "beta"}}) {
    bar {
      name
    }
  }
}
```

**Comment by johnymontana** [comment]
Closing this as I'm hoping the filter feature addressed your use case. Please reopen if the filter functionality doesn't address your case.
